### PR TITLE
Add support for git commit sha

### PR DESCRIPTION
**Pull request by shashwathi**
Copying solution from tektoncd pipeline: https://github.com/tektoncd/pipeline/pull/555
Credit to @dibbles on the original PR

Fixes https://github.com/knative/build/issues/212

**Release Note**

```release-note
Include support for git source with commit ID for revision
```

### CODE REVIEW

File: test/git-source/1-commitid.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 The Knative Authors
```
Comment (mattmoor-sockpuppet) [inline test/git-source/1-commitid.yaml]:
```suggestion
# Copyright 2019 The Knative Authors
```

**Comment by shashwathi** [comment]
@ImJasonH : Updated e2e test setup to check for SA creation and address bot comment on copyright year

**Pull request merged**